src/Browser.php

```diff
@@ -169,50 +165,102 @@ public function getVersion()
     }
 
     /**
-     * Set the Browser to be a robot.
-     *
-     * @param bool $isRobot
+     * Detects scripted agents (robots / bots)
+     * Returns a resolved ScriptedAgent object if detected.
+     * Otherwise returns false.
      *
-     * @return $this
+     * @return ScriptedAgent|bool
      */
-    public function setIsRobot($isRobot)
+    public function detectScriptedAgent()
     {
-        $this->isRobot = (bool)$isRobot;
+        $ua = $this->getUserAgent()->getUserAgentString();
+        if (stripos($ua, 'bot') !== false ||
+            stripos($ua, 'spider') !== false ||
+            stripos($ua, 'crawler') !== false ||
+            stripos($ua, 'preview') !== false ||
+            stripos($ua, 'slurp') !== false ||
+            stripos($ua, 'facebookexternalhit') !== false ||
+            stripos($ua, 'mediapartners') !== false ||
+            stripos($ua, 'google-adwords') !== false ||
+            stripos($ua, 'adxvastfetcher') !== false ||
+            stripos($ua, 'adbeat') !== false ||
+            stripos($ua, 'google favicon') !== false ||
+            stripos($ua, 'webdav client') !== false ||
+            stripos($ua, 'metauri api') !== false ||
+            stripos($ua, 'tlsprobe') !== false ||
+            stripos($ua, 'wpif') !== false ||
+            stripos($ua, 'imgsizer') !== false ||
+            stripos($ua, 'netcraft ssl server survey') !== false ||
+            stripos($ua, 'curl/') !== false ||
+            stripos($ua, 'go-http-client/') !== false ||
+            stripos($ua, 'python') !== false ||
+            stripos($ua, 'libwww') !== false ||
+            stripos($ua, 'wget/') !== false ||
+            stripos($ua, 'zgrab/') !== false ||
+            stripos($ua, 'Java/') !== false ||
+            stripos($ua, '() { :;}; /bin/bash -c') !== false ||
+            stripos($ua, 'browsershots') !== false ||
+            stripos($ua, 'magereport') !== false ||
+            stripos($ua, 'ubermetrics-technologies') !== false ||
+            stripos($ua, 'W3C') !== false ||
+            stripos($ua, 'Validator') !== false ||
+            stripos($ua, 'Jigsaw/') !== false ||
+            stripos($ua, 'bing') !== false ||
+            stripos($ua, 'msn') !== false ||
+            stripos($ua, 'Google Web Preview') !== false ||
+            stripos($ua, 'ips-agent') !== false ||
+            (stripos($ua, 'Chrome/51.0.2704.103') !== false && !isset($_SERVER['HTTP_UPGRADE_INSECURE_REQUESTS']) && stristr($_SERVER['HTTP_ACCEPT_LANGUAGE'], "ru-RU") !== false) //ICQ Preview
```

Comment:
We could still check from the browser class if it is a scripted agent by checking the result of ScriptedAgentDetector. I think it would be cleaner to separate the two and link them with methods instead of having this check in the browser class.